### PR TITLE
DNF-2.0: Merge with dnf-plugin-system-upgrade

### PR DESCRIPTION
**Pull request by stepasm**
Merge of project dnf-plugin-system-upgrade into project dnf-plugins-extras with same license.

### CODE REVIEW

**Comment by stepasm** [comment]
Old system-upgrade works fine when upgrading from fedora:23 to fedora:24 but fails when upgrading fedora:24 to fedora:25 or rawhide. Same issue is with modified system-upgrade. (tested only in VM)
New system upgrade requires dnf-2.0 -> can not be tested with old python API 3.4 on fedora:23.

**Comment by ignatenkobrain** [comment]
@stepasm you can install dnf-2.0 from our nightly repo on F23.

File: plugins/system_upgrade.py [hunk 1]
```diff
@@ -472,7 +472,10 @@ def run_upgrade(self, extcmds):
                 rpms.append(os.path.join(self.state.datadir, f))
         self.base.add_remote_rpms(rpms)
         # set up the upgrade transaction
-        self.base.upgrade_all()
+        if self.opts.distro_sync:
+            self.base.distro_sync()
```
Comment (stepasm) [inline plugins/system_upgrade.py]: distro_sync is mysteriously working only when rootpath is not passed as parameter to Sack.

File: dnf-plugins-extras.spec [hunk 2]
```diff
@@ -1,432 +1,369 @@
...
 BuildArch:      noarch
+
 BuildRequires:  cmake
-BuildRequires:  gettext
-# py2
-%if 0%{?fedora} < 23
-BuildRequires:  python-dnf >= %{dnf_lowest_compatible}
-BuildRequires:  python-dnf < %{dnf_not_compatible}
-%else
+BuildRequires:  gettext pkgconfig
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: There is BR: pkgconfig(systemd)

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,7 +19,23 @@ MESSAGE(STATUS "Python install dir is ${PYTHON_INSTALL_DIR}")
...
+
+if (NOT SYSTEMD_DIR)
+  set (SYSTEMD_DIR /usr/lib/systemd/system)
+endif ()
+
 ADD_SUBDIRECTORY (doc)
 ADD_SUBDIRECTORY (etc)
 ADD_SUBDIRECTORY (plugins)
 ADD_SUBDIRECTORY (po)
+
```
Comment (ignatenkobrain) [inline CMakeLists.txt]: not needed ;)

File: dnf-plugins-extras.spec [hunk 4]
```diff
@@ -1,432 +1,370 @@
...
-
-%description -n python3-dnf-plugins-extras-rpmconf
-RpmConf Plugin for DNF, Python 3 version. Handles .rpmnew, .rpmsave every
-transaction.
+Requires:       python3-%{name}-common = %{?epoch:%{?epoch}:}%{version}-%{release}
+%{?python_provide:%python_provide python3-%{name}-rpmconf}
+Provides:       dnf-command(rpmconf)
+Provides:       %{name}-rpmconf = %{?epoch:%{epoch}:}%{version}-%{release}
+
+BuildRequires:  python3-rpmconf
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: Missing Req: python3-rpmconf

File: dnf-plugins-extras.spec [hunk 5]
```diff
@@ -1,432 +1,370 @@
...
+Summary:        Repomanage Plugin for DNF
+Requires:       python3-%{name}-common = %{?epoch:%{?epoch}:}%{version}-%{release}
 Requires:       python3-dbus
 Requires:       snapper
-%if 0%{?fedora} >= 23
-Provides:       dnf-plugins-extras-snapper = %{version}-%{release}
-Obsoletes:      dnf-plugins-extras-snapper <= 0.0.4-2
-%endif
+%{?python_provide:%python_provide python3-%{name}-snapper}
+Provides:       dnf-command(snapper)
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: that's not true

File: dnf-plugins-extras.spec [hunk 6]
```diff
@@ -1,432 +1,370 @@
...
-%description -n python3-dnf-plugins-extras-torproxy
-Torproxy plugin force dnf to use tor to download packages. It make sure that
-tor is working and avoid leaking hostname by using the proper sock5 interface.
+Obsoletes:	fedup < 0.9.4
+Obsoletes:	dnf-plugin-system-upgrade < 0.10
+Obsoletes:	python3-dnf-plugin-system-upgrade < 0.10
+
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  systemd
+BuildRequires:  python2-systemd
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: Missing Req: python2-systemd

File: dnf-plugins-extras.spec [hunk 7]
```diff
@@ -1,432 +1,370 @@
...
+Summary:        System Upgrade Plugin for DNF
+Requires:       python2-%{name}-common = %{?epoch:%{?epoch}:}%{version}-%{release}
+%{?python_provide:%python_provide python2-%{name}-system-upgrade}
 
-%description -n python3-dnf-plugins-extras-torproxy
-Torproxy plugin force dnf to use tor to download packages. It make sure that
-tor is working and avoid leaking hostname by using the proper sock5 interface.
+Obsoletes:	fedup < 0.9.4
+Obsoletes:	dnf-plugin-system-upgrade < 0.10
+Obsoletes:	python3-dnf-plugin-system-upgrade < 0.10
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: python2-* shouldn't obsolete python3-*

File: dnf-plugins-extras.spec [hunk 8]
```diff
@@ -1,432 +1,370 @@
...
+%package -n python2-%{name}-system-upgrade
+Summary:        System Upgrade Plugin for DNF
+Requires:       python2-%{name}-common = %{?epoch:%{?epoch}:}%{version}-%{release}
+%{?python_provide:%python_provide python2-%{name}-system-upgrade}
 
-%description -n python3-dnf-plugins-extras-torproxy
-Torproxy plugin force dnf to use tor to download packages. It make sure that
-tor is working and avoid leaking hostname by using the proper sock5 interface.
+Obsoletes:	fedup < 0.9.4
+Obsoletes:	dnf-plugin-system-upgrade < 0.10
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: I think this should be moved to py3 subpkg

File: dnf-plugins-extras.spec [hunk 9]
```diff
@@ -1,432 +1,370 @@
...
+Requires:       python3-%{name}-common = %{?epoch:%{?epoch}:}%{version}-%{release}
+%{?python_provide:%python_provide python3-%{name}-system-upgrade}
+Provides:       dnf-command(system-upgrade)
+Provides:       %{name}-system-upgrade = %{?epoch:%{epoch}:}%{version}-%{release}
+
+Obsoletes:	fedup < 0.9.4
+
+Obsoletes:	dnf-plugin-system-upgrade < 0.10
+Obsoletes:	python3-dnf-plugin-system-upgrade < 0.10
+Conflicts:	PackageKit < 1.0.8
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: Useless

File: dnf-plugins-extras.spec [hunk 10]
```diff
@@ -1,432 +1,370 @@
...
+
+BuildRequires:  pkgconfig(systemd)
+BuildRequires:  systemd
+BuildRequires:  python3-systemd
```
Comment (ignatenkobrain) [inline dnf-plugins-extras.spec]: Missing Req: python3-systemd

**Comment by Conan-Kudo** [comment]
@stepasm Can you please rebase against master and fix the conflicts?

**Comment by Conan-Kudo** [comment]
I'm not particularly happy with how much unrelated changes have been made in this PR.

@stepasm Can you please split things up so that we have PRs specifically for merging in dnf-system-upgrade into this repository and the other stuff you've done (torproxy, refactoring the spec, etc.)?

It's just too much...

**Comment by jsilhan** [comment]
this PR is replaced by #73

**Pull request closed without merging**